### PR TITLE
 Optimize ClientBuilderImpl

### PR DESCRIPTION
**Pull request by yws-tracy**
Optimize ClientBuilderImpl        
          1. unify argument check logic using Preconditions instead of partly using Preconditions partly use if ...
          2. simplify if else logic on checking serviceUrlProvider

*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

I  find `ClientBuilderImpl` class can be improved by unify code style when checking argument validity and simplifying  if else logic for  param serviceUrlProvider verifying.   At present,  in `ClientBuilderImpl` some params verifying use Preconditions ， while some do not, it is not unified，and some if else logic can be omitted eg: when checking serviceUrlProvider.  

therefore, I submit this pr,  hope to be adopted.  thanks 

### Modifications

Unify params verifying logic in `ClientBuilderImpl` use Preconditions instead of  if else,   does not change code logic itself. 
also omit if else when using Preconditions to check param serviceUrlProvider to simplify code logic 

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java [hunk 1]
```diff
@@ -339,7 +328,8 @@ public ClientBuilder enableTransaction(boolean enableTransaction) {
 
     @Override
     public ClientBuilder dnsLookupBind(String address, int port) {
-        checkArgument(port >= 0 && port <= 65535, "DnsLookBindPort need to be within the range of 0 and 65535");
+        checkArgument(port >= 0 && port <= 65535,
```
Comment (Shoothzj) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:331]: is that a unnecessary format?

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java [hunk 2]
```diff
@@ -79,15 +75,13 @@ public ClientBuilder clone() {
     @Override
     public ClientBuilder loadConf(Map<String, Object> config) {
         conf = ConfigurationDataUtils.loadData(
-            config, conf, ClientConfigurationData.class);
+                config, conf, ClientConfigurationData.class);
```
Comment (Shoothzj) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:78]: is that a unnecessary format?

**Review by Shoothzj — CHANGES_REQUESTED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java (hunk 1 above)
Comment (yws-tracy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:331]: it's too long,  I split into two line by the way

Comment (yws-tracy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:331]: do I need to recover ?

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java (hunk 2 above)
Comment (yws-tracy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:78]: yeah,  I recover to normal

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java (hunk 1 above)
Comment (yws-tracy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java]: recovered!

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java (hunk 2 above)
Comment (yws-tracy) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java]: recovered

**Review by Jason918 — APPROVED** [review]
LGTM

**Review by Shoothzj — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java [hunk 3]
```diff
@@ -323,9 +312,8 @@ public ClientBuilder clock(Clock clock) {
 
     @Override
     public ClientBuilder proxyServiceUrl(String proxyServiceUrl, ProxyProtocol proxyProtocol) {
-        if (StringUtils.isNotBlank(proxyServiceUrl) && proxyProtocol == null) {
-            throw new IllegalArgumentException("proxyProtocol must be present with proxyServiceUrl");
-        }
+        checkArgument(StringUtils.isBlank(proxyServiceUrl) || proxyProtocol != null,
+                "proxyProtocol must be present with proxyServiceUrl");
```
Comment (Technoboy-) [inline pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:316]: Ah ?  is this equals ?

Comment (Shoothzj) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:316]: @Technoboy- Yes, the checkArgument method's arguement should be true, so reversed

Comment (Technoboy-) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:316]: This makes the code unreadable.

**Review by RobertIndie — APPROVED** [review]

File: pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java (hunk 3 above)
Comment (codelipenghui) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:316]: What is the expected behavior here? Both the `proxyServiceUrl` and `proxyProtocol` can not be null or empty?

Comment (RobertIndie) [reply pulsar-client/src/main/java/org/apache/pulsar/client/impl/ClientBuilderImpl.java:316]: @codelipenghui I think the expected behavior is to check that the `proxyProtocol` must be provided if the `proxyServiceUrl` is provided. I agree with @Technoboy- . I think we could make this more readable

**Pull request merged**